components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -54,4 +58,59 @@ function AutomaticPointsTable:parseTournaments(args)
 	return tournaments
 end
 
+function AutomaticPointsTable:parseTeams(args, tournamentCount)
+	local teams = {}
+	for _, team in Table.iter.pairsByPrefix(args, 'team') do
+		local parsedTeam = Json.parse(team)
+		parsedTeam.aliases = self:parseAliases(parsedTeam, tournamentCount)
+		parsedTeam.deductions = self:parseDeductions(parsedTeam, tournamentCount)
+		parsedTeam.manualPoints = self:parseManualPoints(parsedTeam, tournamentCount)
+		table.insert(teams, parsedTeam)
+	end
+	return teams
+end
+
+--- Parses the team aliases, used in cases where a team is picked up by an org or changed name in some
+--- of the tournaments, in which case aliases are required to correctly query the team's results & points
+function AutomaticPointsTable:parseAliases(team, tournamentCount)
+	local aliases = {}
+	for index = 1, tournamentCount do
+		if String.isNotEmpty(team['alias' .. index]) then
+			aliases[index] = team['alias' .. index]
+		end
+	end
+	return aliases
+end
+
+--- Parses the teams' deductions, used in cases where a team has disbanded or made a roster change
+--- that causes them to lose a portion or all of their points that they've accumulated up until that change
+function AutomaticPointsTable:parseDeductions(team, tournamentCount)
+	local deductions = {}
+	for index = 1, tournamentCount do
+		if String.isNotEmpty(team['deduction' .. index .. 'note']) then
+			if not deductions[index] then
+				deductions[index] = {}
+			end
+			deductions[index].note = team['deduction' .. index .. 'note']
+		end
+		if String.isNotEmpty(team['deduction' .. index]) then
+			if not deductions[index] then
+				deductions[index] = {}
+			end
+			deductions[index].amount = tonumber(team['deduction' .. index])
+		end
+	end

```

Comment:
Is it planned for `deductionX` being a requirement to use `deducationXnote`? (ie just setting a deducationXnote without the corrosponding dedudctionX would do nothing in practice)
**If so**, this could be rewriten as 
```suggestion
	for index = 1, tournamentCount do
		if String.isNotEmpty(team['deduction' .. index]) then
			if not deductions[index] then
				deductions[index] = {}
			end
			deductions[index].amount = tonumber(team['deduction' .. index])

			if String.isNotEmpty(team['deduction' .. index .. 'note']) then
				deductions[index].note = team['deduction' .. index .. 'note']
			end
		end
	end

```